dcos/marathon.py

```diff
@@ -109,8 +109,14 @@ def response_error_message(cls, status_code, reason, request_method,
             return template.format(request_method, request_url, reason, suffix)
 
         if status_code == 409:
-            return ('App, group, or pod is locked by one or more deployments. '
-                    'Override with --force.')
+            _, _, path, _, _, _ = urllib.parse.urlparse(request_url)
+            for resource_name in ['app', 'group', 'pod']:
+                if '/v2/{}s'.format(resource_name) in path:
```

Comment:
I think it's good enough if we just look for the resource in the path. We know this is a marathon url and only one resource is in their endpoint names. 

we can then do something like `resource_name = next((resource for resource in ['app', 'group', 'pod'] if resource in path), 'Resource')`

https://docs.python.org/3.4/library/functions.html#next